### DISCUSSION TITLE
🕶️ middleware 설정

### DISCUSSION THREAD

**Discussion by soryeongk** [TeamBookTez/nextjs-book-stairs#28] [Ideas]
로그인 로직을 체크하는 middleware를 설정해보려합니다. 어떤 방식이 좋을지 comment를 남겨주세요!
@joohaem @Gyuminn  

# 1. Next.js Beta 버전의 미들웨어

[공식문서 링크(클릭)](https://github.com/vercel/examples/tree/main/edge-functions/basic-auth-password)
pages 폴더 안에 _middleware.ts를 만들어두면 모든 페이지에 대해 이동마다 middleware.ts가 가장 먼저 실행됩니다.

# 2. Recoil seletor

[공식문서 링크(클릭)](https://recoiljs.org/ko/docs/guides/asynchronous-data-queries/)
redux에서는 제공하는 middleware가 recoil에는 없다..! 아직까지 리코일이 리덕스를 대체하지 못하는 몇 가지 이유 중 하나!

미들웨어로 뭘 하고 싶냐면,

유저가 페이지를 탐색할 때 이 사람이 로그인을 했는지 안했는지를 판별하는 것을 개선하고 싶어! recoil selector로 비동기처리를 해서 사용하는 것이 가능하다!

지금 훅을 만들어서 쓰는 것은 로그인 여부를 판별하는 useCheckLoginState를 실행하고 리코일 변수를 업데이트하는 형식이라 로그인 로직 하나 검사하는데 커스텀훅과 useEffect가 모두 사용되는 중..! selector를 쓰면 더 보기 쉽고 리코일 공부해보기도 좋을 것 같아!

# 3. SWR

[공식문서 링크(클릭](https://swr.vercel.app/examples/auth)

api-auth와 같은 key로 관리되는 useSWR을 하나 만들어서 로그인 판별이 필요한 페이지에 넣어두는거지. swr을 사용하면 revalidate 타이밍을 설정하는 것이나 mutation하는 것도 간편해질 것 같은데, 우리 서비스가 그렇게 자주 로그인 여부를 판별해야하는 서비스인가?에 대해서는 고민해봐야할 것 같아!

**Comment by joohaem** [comment]
사전에 논의되었던 자료 첨부합니다!

**📎너소서 헤더 처리 ::** https://github.com/Neogasogaeseo/Naega-Web/blob/dev/src/infrastructure/remote/base.ts
